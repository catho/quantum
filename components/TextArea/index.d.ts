import React from 'react';

export interface TextAreaProps {
    disabled?: boolean;
    error?: string;
    helperText?: string;
    label?: string;
    onChange?: React.ChangeEventHandler<HTMLInputElement>;
    placeholder?: string;
    required?: boolean;
    value?: string;
    id?: string | number;
}

export default class TextArea extends React.Component<TextAreaProps> {}
