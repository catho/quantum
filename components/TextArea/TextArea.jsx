import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { FieldGroup, uniqId } from '../shared';
import {
  InputLabel,
  InputErrorMessage,
  HelperText,
  RequiredMark,
  TextInput,
} from '../Input/sub-components';

const ID_GENERATOR = uniqId('textarea-');

const TextAreaTag = styled(TextInput)`
  display: block;
  min-height: 108px;
  margin-top: 8px;
  transition: border 0.2s ease-in-out, box-shadow 0.2s ease-in-out;
`;

class TextArea extends React.Component {
  constructor(props) {
    super(props);

    const { id, value } = props;

    this.state = {
      hasDefaultValue: value !== null && value[0],
      currentValue: value,
    };

    this._id = id || ID_GENERATOR.next().value;
  }

  onChangeTextArea = ev => {
    const { onChange } = this.props;
    const inputValue = ev.currentTarget.value;
    this.setState({
      currentValue: inputValue,
      hasDefaultValue: null,
    });

    onChange(ev);
  };

  render() {
    const { label, required, helperText, error, id, ...rest } = this.props;
    const { hasDefaultValue, currentValue } = this.state;

    return (
      <FieldGroup>
        {label && (
          <InputLabel htmlFor={this._id}>
            {label}
            {required && <RequiredMark>*</RequiredMark>}
          </InputLabel>
        )}
        <TextAreaTag
          {...rest}
          hasDefaultValue={hasDefaultValue}
          value={currentValue}
          error={error}
          id={this._id}
          required={required}
          as="textarea"
          onChange={this.onChangeTextArea}
        />
        {helperText && <HelperText>{helperText}</HelperText>}
        {error && (
          <InputErrorMessage helperText={helperText}>{error}</InputErrorMessage>
        )}
      </FieldGroup>
    );
  }
}

TextArea.defaultProps = {
  disabled: false,
  error: '',
  helperText: '',
  label: undefined,
  onChange: () => {},
  placeholder: '',
  required: false,
  value: '',
  id: undefined,
};

TextArea.propTypes = {
  disabled: PropTypes.bool,
  error: PropTypes.string,
  helperText: PropTypes.string,
  label: PropTypes.string,
  onChange: PropTypes.func,
  placeholder: PropTypes.string,
  required: PropTypes.bool,
  value: PropTypes.string,
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

TextAreaTag.displayName = 'TextAreaTag';

export default TextArea;
