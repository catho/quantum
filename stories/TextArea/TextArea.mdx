import { Meta, Canvas, Story, ArgsTable } from '@storybook/addon-docs';
import { Header } from '../shared';
import TextArea from '../../components/TextArea';
import { Default, Disabled, Required, WithAutoSize, WithCustomAutoSize, WithHelperText, WithError, WithLabel } from './TextArea.stories.jsx';

<Meta title="Forms/TextArea" component={TextArea}  />

<Header title="TextArea">
    The TextArea component is used to provide a text input field with multiple lines in a form. It is defined with the textarea tag and can hold an unlimited number of characters.
</Header>

- [Importing and usage](#importing)
- [API](#api)
- [Examples](#examples)

## <a id="importing"></a>Importing and usage

```js
import TextArea from '@catho/quantum/TextArea';
```

<Canvas>
  <Story of={Default}/>
</Canvas>

## <a id="api"></a>API

<ArgsTable of={TextArea} />

## <a id="examples"></a>Examples

### Required

<Canvas>
  <Story of={Required}/>
</Canvas>

### Disabled

<Canvas>
  <Story of={Disabled}/>
</Canvas>

### With Auto Size

<Canvas>
  <Story of={WithAutoSize}/>
</Canvas>

### With Custom Auto Size

<Canvas>
  <Story of={WithCustomAutoSize}/>
</Canvas>


### With Helper Text

<Canvas>
  <Story of={WithHelperText}/>
</Canvas>

### With Label

<Canvas>
  <Story of={WithLabel}/>
</Canvas>

### With Error

<Canvas>
  <Story of={WithError}/>
</Canvas>