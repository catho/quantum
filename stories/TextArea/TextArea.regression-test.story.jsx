import React from 'react';
import { storiesOf } from '@storybook/react';

import TextArea from '../../components/TextArea';

storiesOf('TextArea', module)
  .add('default', () => <TextArea />)
  .add('disabled', () => <TextArea disabled />)
  .add('pre valued', () => <TextArea value="text value" />)
  .add('label', () => <TextArea label="label" />)
  .add('required', () => <TextArea label="label" required />)
  .add('helperText', () => <TextArea helperText="helperText" />)
  .add('placeholder', () => <TextArea placeholder="placeholder" />);
