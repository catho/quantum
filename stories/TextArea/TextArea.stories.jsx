import TextArea from '../../components/TextArea';

export default {
  title: 'Forms/TextArea',
  component: TextArea,
};

const Template = (args) => <TextArea {...args} />;

export const Default = Template.bind({});

export const Disabled = Template.bind({});
Disabled.args = {
  disabled: true,
};

export const Required = Template.bind({});
Required.args = {
  required: true,
  label: 'Some text label',
};

export const WithAutoSize = Template.bind({});
WithAutoSize.args = {
  isAutoResize: true,
};

export const WithCustomAutoSize = Template.bind({});
WithCustomAutoSize.args = {
  isAutoResize: true,
  autoResizeConfig: { minRows: 2, initialRows: 2, maxRows: 3 },
};

export const WithHelperText = Template.bind({});
WithHelperText.args = {
  helperText: 'Some text...',
};

export const WithError = Template.bind({});
WithError.args = {
  error: 'Some error text',
};

export const WithLabel = Template.bind({});
WithLabel.args = {
  label: 'Some text label...',
};
