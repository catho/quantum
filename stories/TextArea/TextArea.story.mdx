import { Meta, Canvas, Story, ArgsTable } from '@storybook/addon-docs';
import { Header } from '../shared';
import TextArea from '../../components/TextArea';
import { Default, Disabled, Required, WithAutoSize, WithCustomAutoSize, WithHelperText, WithError, WithLabel } from './TextArea.stories.jsx';

<Meta title="Forms/TextArea" component={TextArea}  />

<Header title="TextArea">
    The TextArea component is used to provide a text input field with multiple lines in a form. It is defined with the textarea tag and can hold an unlimited number of characters.
</Header>

- [Importing and usage](#importing)
- [API](#api)
- [Examples](#examples)

## <a id="importing"></a>Importing and usage

```js
import TextArea from '@catho/quantum/TextArea';
```

<Canvas>
  <Story story={Default}/>
</Canvas>

## <a id="api"></a>API

<ArgsTable of={TextArea} />

## <a id="examples"></a>Examples

### Required

<Canvas>
  <Story story={Required}/>
</Canvas>

### Disabled

<Canvas>
  <Story story={Disabled}/>
</Canvas>

### With Auto Size

<Canvas>
  <Story story={WithAutoSize}/>
</Canvas>

### With Custom Auto Size

<Canvas>
  <Story story={WithCustomAutoSize}/>
</Canvas>


### With Helper Text

<Canvas>
  <Story story={WithHelperText}/>
</Canvas>

### With Label

<Canvas>
  <Story story={WithLabel}/>
</Canvas>

### With Error

<Canvas>
  <Story story={WithError}/>
</Canvas>