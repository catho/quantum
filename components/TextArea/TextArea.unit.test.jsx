import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import TextArea from './TextArea';

describe('TextArea component', () => {
  it('should match snapshots', () => {
    const TEXT_AREAS = [
      <TextArea />,
      <TextArea value="foo" />,
      <TextArea label="Text label" />,
      <TextArea label="Text label" required />,
      <TextArea disabled />,
      <TextArea placeholder="this input has a placeholder" />,
      <TextArea helperText="this is a helper text" />,
      <TextArea error="Error message" />,
    ];

    TEXT_AREAS.forEach(textArea =>
      expect(renderer.create(textArea).toJSON()).toMatchSnapshot(),
    );
  });

  it('should has a required signal when "required" and "label" props are set ', () => {
    const component = shallow(<TextArea label="Text label" required />);
    expect(component.find('InputLabel').text()).toMatch('*');
  });

  it('should have a placeholder when "placeholder" prop is set ', () => {
    const placeholderText = 'this input has a placeholder';
    const component = shallow(<TextArea placeholder={placeholderText} />);
    const textAreaPlaceholder = component
      .find('TextAreaTag')
      .prop('placeholder');

    expect(textAreaPlaceholder).toMatch(placeholderText);
  });

  it('should have a helper text when "helper text" prop is set ', () => {
    const helperTextContent = 'this is a helper text';
    const component = shallow(<TextArea helperText={helperTextContent} />);
    const textAreaHelperText = component.find('HelperText').text();

    expect(textAreaHelperText).toMatch(helperTextContent);
  });

  it('should have a error text when "error" prop is set', () => {
    const errorMessageContent = 'Error message';
    const component = shallow(<TextArea error={errorMessageContent} />);
    const errorMessage = component.find('InputErrorMessage').text();
    expect(errorMessage).toMatch(errorMessageContent);
  });

  it('should onChange props be called', () => {
    const onChangeEventMock = jest.fn();
    const wrapper = mount(
      <TextArea label="Text label" value="foo" onChange={onChangeEventMock} />,
    );
    const componentText = wrapper.find('TextAreaTag');
    componentText.simulate('change');
    expect(onChangeEventMock).toHaveBeenCalled();
  });

  describe('with a label', () => {
    it('should match label "htmlFor" label param with "id" input param', () => {
      const id = 'input-id';
      const wrapper = shallow(<TextArea label="Text label" id={id} />);
      const input = wrapper.find('TextAreaTag');
      const label = wrapper.find('InputLabel');
      const labelHtmlFor = label.prop('htmlFor');
      const inputId = input.prop('id');

      expect(labelHtmlFor).toEqual(id);
      expect(inputId).toEqual(id);
    });

    it('should match label "htmlFor" label param and "input" param with generated id', () => {
      const wrapper = shallow(<TextArea label="Text label" value="foo" />);
      const labelId = wrapper.find('InputLabel').prop('htmlFor');
      const inputId = wrapper.find('TextAreaTag').prop('id');

      expect(labelId).toEqual(inputId);
    });
  });
});
