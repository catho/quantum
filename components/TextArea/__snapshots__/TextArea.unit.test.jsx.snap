// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea component should match snapshots 1`] = `
.c0 {
  position: relative;
  margin-bottom: 20px;
  min-width: 250px;
  width: 100%;
}

.c1 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  background-color: #ffffff;
  border: 2px solid #999999;
  color: #424242;
  margin-top: 8px;
  padding: 8px 16px;
  display: block;
  min-height: 108px;
  margin-top: 8px;
  -webkit-transition: border 0.2s ease-in-out,box-shadow 0.2s ease-in-out;
  transition: border 0.2s ease-in-out,box-shadow 0.2s ease-in-out;
}

.c1::-webkit-inner-spin-button,
.c1::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c1::-webkit-calendar-picker-indicator,
.c1::-webkit-search-cancel-button {
  display: none;
}

.c1::-ms-clear {
  display: none;
}

.c1:hover,
.c1:focus {
  border-color: #1250C4;
  box-shadow: 0px 3px 1px -2px rgba(18,80,196,0.2),0px 2px 2px 0px rgba(18,80,196,0.14),0px 1px 5px 0px rgba(18,80,196,0.12);
}

.c1[disabled] {
  background-color: #f2f2f2;
  border-color: #999999;
  box-shadow: none;
  color: #999999;
  cursor: not-allowed;
}

.c1:-webkit-autofill {
  box-shadow: 0 0 0 1000px #ffe5e9 inset;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    disabled={false}
    id="textarea-0"
    onChange={[Function]}
    placeholder=""
    required={false}
    value=""
  />
  
  
</div>
`;

exports[`TextArea component should match snapshots 2`] = `
.c0 {
  position: relative;
  margin-bottom: 20px;
  min-width: 250px;
  width: 100%;
}

.c1 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  background-color: #ffffff;
  border: 2px solid #999999;
  color: #424242;
  margin-top: 8px;
  padding: 8px 16px;
  background-color: #E5EDFC;
  display: block;
  min-height: 108px;
  margin-top: 8px;
  -webkit-transition: border 0.2s ease-in-out,box-shadow 0.2s ease-in-out;
  transition: border 0.2s ease-in-out,box-shadow 0.2s ease-in-out;
}

.c1::-webkit-inner-spin-button,
.c1::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c1::-webkit-calendar-picker-indicator,
.c1::-webkit-search-cancel-button {
  display: none;
}

.c1::-ms-clear {
  display: none;
}

.c1:hover,
.c1:focus {
  border-color: #1250C4;
  box-shadow: 0px 3px 1px -2px rgba(18,80,196,0.2),0px 2px 2px 0px rgba(18,80,196,0.14),0px 1px 5px 0px rgba(18,80,196,0.12);
}

.c1[disabled] {
  background-color: #f2f2f2;
  border-color: #999999;
  box-shadow: none;
  color: #999999;
  cursor: not-allowed;
}

.c1:-webkit-autofill {
  box-shadow: 0 0 0 1000px #ffe5e9 inset;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    disabled={false}
    id="textarea-1"
    onChange={[Function]}
    placeholder=""
    required={false}
    value="foo"
  />
  
  
</div>
`;

exports[`TextArea component should match snapshots 3`] = `
.c0 {
  position: relative;
  margin-bottom: 20px;
  min-width: 250px;
  width: 100%;
}

.c1 {
  display: block;
  font-weight: bold;
  margin: 8px 0 0;
}

.c2 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  background-color: #ffffff;
  border: 2px solid #999999;
  color: #424242;
  margin-top: 8px;
  padding: 8px 16px;
  display: block;
  min-height: 108px;
  margin-top: 8px;
  -webkit-transition: border 0.2s ease-in-out,box-shadow 0.2s ease-in-out;
  transition: border 0.2s ease-in-out,box-shadow 0.2s ease-in-out;
}

.c2::-webkit-inner-spin-button,
.c2::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c2::-webkit-calendar-picker-indicator,
.c2::-webkit-search-cancel-button {
  display: none;
}

.c2::-ms-clear {
  display: none;
}

.c2:hover,
.c2:focus {
  border-color: #1250C4;
  box-shadow: 0px 3px 1px -2px rgba(18,80,196,0.2),0px 2px 2px 0px rgba(18,80,196,0.14),0px 1px 5px 0px rgba(18,80,196,0.12);
}

.c2[disabled] {
  background-color: #f2f2f2;
  border-color: #999999;
  box-shadow: none;
  color: #999999;
  cursor: not-allowed;
}

.c2:-webkit-autofill {
  box-shadow: 0 0 0 1000px #ffe5e9 inset;
}

<div
  className="c0"
>
  <label
    className="c1"
    htmlFor="textarea-2"
  >
    Text label
  </label>
  <textarea
    className="c2"
    disabled={false}
    id="textarea-2"
    onChange={[Function]}
    placeholder=""
    required={false}
    value=""
  />
  
  
</div>
`;

exports[`TextArea component should match snapshots 4`] = `
.c0 {
  position: relative;
  margin-bottom: 20px;
  min-width: 250px;
  width: 100%;
}

.c1 {
  display: block;
  font-weight: bold;
  margin: 8px 0 0;
}

.c2 {
  color: #ec584e;
}

.c3 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  background-color: #ffffff;
  border: 2px solid #999999;
  color: #424242;
  margin-top: 8px;
  padding: 8px 16px;
  display: block;
  min-height: 108px;
  margin-top: 8px;
  -webkit-transition: border 0.2s ease-in-out,box-shadow 0.2s ease-in-out;
  transition: border 0.2s ease-in-out,box-shadow 0.2s ease-in-out;
}

.c3::-webkit-inner-spin-button,
.c3::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c3::-webkit-calendar-picker-indicator,
.c3::-webkit-search-cancel-button {
  display: none;
}

.c3::-ms-clear {
  display: none;
}

.c3:hover,
.c3:focus {
  border-color: #1250C4;
  box-shadow: 0px 3px 1px -2px rgba(18,80,196,0.2),0px 2px 2px 0px rgba(18,80,196,0.14),0px 1px 5px 0px rgba(18,80,196,0.12);
}

.c3[disabled] {
  background-color: #f2f2f2;
  border-color: #999999;
  box-shadow: none;
  color: #999999;
  cursor: not-allowed;
}

.c3:-webkit-autofill {
  box-shadow: 0 0 0 1000px #ffe5e9 inset;
}

<div
  className="c0"
>
  <label
    className="c1"
    htmlFor="textarea-3"
  >
    Text label
    <em
      className="c2"
    >
      *
    </em>
  </label>
  <textarea
    className="c3"
    disabled={false}
    id="textarea-3"
    onChange={[Function]}
    placeholder=""
    required={true}
    value=""
  />
  
  
</div>
`;

exports[`TextArea component should match snapshots 5`] = `
.c0 {
  position: relative;
  margin-bottom: 20px;
  min-width: 250px;
  width: 100%;
}

.c1 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  background-color: #ffffff;
  border: 2px solid #999999;
  color: #424242;
  margin-top: 8px;
  padding: 8px 16px;
  display: block;
  min-height: 108px;
  margin-top: 8px;
  -webkit-transition: border 0.2s ease-in-out,box-shadow 0.2s ease-in-out;
  transition: border 0.2s ease-in-out,box-shadow 0.2s ease-in-out;
}

.c1::-webkit-inner-spin-button,
.c1::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c1::-webkit-calendar-picker-indicator,
.c1::-webkit-search-cancel-button {
  display: none;
}

.c1::-ms-clear {
  display: none;
}

.c1:hover,
.c1:focus {
  border-color: #1250C4;
  box-shadow: 0px 3px 1px -2px rgba(18,80,196,0.2),0px 2px 2px 0px rgba(18,80,196,0.14),0px 1px 5px 0px rgba(18,80,196,0.12);
}

.c1[disabled] {
  background-color: #f2f2f2;
  border-color: #999999;
  box-shadow: none;
  color: #999999;
  cursor: not-allowed;
}

.c1:-webkit-autofill {
  box-shadow: 0 0 0 1000px #ffe5e9 inset;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    disabled={true}
    id="textarea-4"
    onChange={[Function]}
    placeholder=""
    required={false}
    value=""
  />
  
  
</div>
`;

exports[`TextArea component should match snapshots 6`] = `
.c0 {
  position: relative;
  margin-bottom: 20px;
  min-width: 250px;
  width: 100%;
}

.c1 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  background-color: #ffffff;
  border: 2px solid #999999;
  color: #424242;
  margin-top: 8px;
  padding: 8px 16px;
  color: #999999;
  display: block;
  min-height: 108px;
  margin-top: 8px;
  -webkit-transition: border 0.2s ease-in-out,box-shadow 0.2s ease-in-out;
  transition: border 0.2s ease-in-out,box-shadow 0.2s ease-in-out;
}

.c1::-webkit-inner-spin-button,
.c1::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c1::-webkit-calendar-picker-indicator,
.c1::-webkit-search-cancel-button {
  display: none;
}

.c1::-ms-clear {
  display: none;
}

.c1:hover,
.c1:focus {
  border-color: #1250C4;
  box-shadow: 0px 3px 1px -2px rgba(18,80,196,0.2),0px 2px 2px 0px rgba(18,80,196,0.14),0px 1px 5px 0px rgba(18,80,196,0.12);
}

.c1[disabled] {
  background-color: #f2f2f2;
  border-color: #999999;
  box-shadow: none;
  color: #999999;
  cursor: not-allowed;
}

.c1:-webkit-autofill {
  box-shadow: 0 0 0 1000px #ffe5e9 inset;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    disabled={false}
    id="textarea-5"
    onChange={[Function]}
    placeholder="this input has a placeholder"
    required={false}
    value=""
  />
  
  
</div>
`;

exports[`TextArea component should match snapshots 7`] = `
.c0 {
  position: relative;
  margin-bottom: 20px;
  min-width: 250px;
  width: 100%;
}

.c2 {
  color: #ac001a;
  margin-top: 8px;
  cursor: text;
  display: block;
  line-height: normal;
  font-size: 14px;
  font-style: italic;
  font-weight: 600;
  color: #424242;
}

.c1 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  background-color: #ffffff;
  border: 2px solid #999999;
  color: #424242;
  margin-top: 8px;
  padding: 8px 16px;
  display: block;
  min-height: 108px;
  margin-top: 8px;
  -webkit-transition: border 0.2s ease-in-out,box-shadow 0.2s ease-in-out;
  transition: border 0.2s ease-in-out,box-shadow 0.2s ease-in-out;
}

.c1::-webkit-inner-spin-button,
.c1::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c1::-webkit-calendar-picker-indicator,
.c1::-webkit-search-cancel-button {
  display: none;
}

.c1::-ms-clear {
  display: none;
}

.c1:hover,
.c1:focus {
  border-color: #1250C4;
  box-shadow: 0px 3px 1px -2px rgba(18,80,196,0.2),0px 2px 2px 0px rgba(18,80,196,0.14),0px 1px 5px 0px rgba(18,80,196,0.12);
}

.c1[disabled] {
  background-color: #f2f2f2;
  border-color: #999999;
  box-shadow: none;
  color: #999999;
  cursor: not-allowed;
}

.c1:-webkit-autofill {
  box-shadow: 0 0 0 1000px #ffe5e9 inset;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    disabled={false}
    id="textarea-6"
    onChange={[Function]}
    placeholder=""
    required={false}
    value=""
  />
  <span
    className="c2"
  >
    this is a helper text
  </span>
  
</div>
`;

exports[`TextArea component should match snapshots 8`] = `
.c0 {
  position: relative;
  margin-bottom: 20px;
  min-width: 250px;
  width: 100%;
}

.c2 {
  color: #ac001a;
  margin-top: 8px;
  cursor: text;
  display: block;
  line-height: normal;
  font-size: 14px;
  font-style: italic;
  font-weight: 600;
}

.c1 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  background-color: #ffffff;
  border: 2px solid #999999;
  color: #424242;
  margin-top: 8px;
  padding: 8px 16px;
  border-color: #e91b0c;
  display: block;
  min-height: 108px;
  margin-top: 8px;
  -webkit-transition: border 0.2s ease-in-out,box-shadow 0.2s ease-in-out;
  transition: border 0.2s ease-in-out,box-shadow 0.2s ease-in-out;
}

.c1::-webkit-inner-spin-button,
.c1::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c1::-webkit-calendar-picker-indicator,
.c1::-webkit-search-cancel-button {
  display: none;
}

.c1::-ms-clear {
  display: none;
}

.c1:hover,
.c1:focus {
  border-color: #e91b0c;
  box-shadow: 0px 3px 1px -2px rgba(233,27,12,0.2),0px 2px 2px 0px rgba(233,27,12,0.14),0px 1px 5px 0px rgba(233,27,12,0.12);
}

.c1[disabled] {
  background-color: #f2f2f2;
  border-color: #999999;
  box-shadow: none;
  color: #999999;
  cursor: not-allowed;
}

.c1:-webkit-autofill {
  box-shadow: 0 0 0 1000px #ffe5e9 inset;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    disabled={false}
    id="textarea-7"
    onChange={[Function]}
    placeholder=""
    required={false}
    value=""
  />
  
  <span
    className="c2"
  >
    Error message
  </span>
</div>
`;
